### PR TITLE
Project problems are resolved in headless environment.

### PR DESCRIPTION
**Pull request by sdedic**
This PR aims to solve issues with newly opened projects in headless environment  (required by LSP server). Project problems UI demand an interactive GUI, which can be partly handled by `DialogDisplayer` implementation for headless env. In an absence of better interaction with the user, all reported problems that can be fixed are attempted to be fixed. Results are reported using StatusDisplayer.

### CODE REVIEW

**Comment by sdedic** [comment]
Reported [NETBEANS-5226](https://issues.apache.org/jira/browse/NETBEANS-5226) for further thoughts.

**Comment by JaroslavTulach** [comment]
I am trying to verify, but so far it is not working for me:
```
"org.netbeans.modules.project.ui.problems.BrokenReferencesImpl"
	at org.netbeans.modules.java.lsp.server.ui.UIContext.<init>(UIContext.java:32)
	at org.netbeans.modules.java.lsp.server.ui.UIContext$LogImpl.<init>(UIContext.java:87)
	at org.netbeans.modules.java.lsp.server.ui.UIContext$LogImpl.<clinit>(UIContext.java:85)
	at org.netbeans.modules.java.lsp.server.ui.UIContext.find(UIContext.java:62)
	at org.netbeans.modules.java.lsp.server.ui.AbstractDialogDisplayer.notify(AbstractDialogDisplayer.java:49)
	at org.netbeans.modules.project.ui.problems.BrokenReferencesImpl$1.run(BrokenReferencesImpl.java:109)
	at org.netbeans.modules.project.ui.problems.BrokenReferencesImpl$2.run(BrokenReferencesImpl.java:143)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1418)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:278)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)
```
e.g. a `LogImpl` UI is used instead. Still investigating.

File: platform/openide.util.ui/src/org/netbeans/modules/openide/util/NbMutexEventProvider.java [hunk 1]
```diff
@@ -24,9 +24,11 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 import org.openide.util.BaseUtilities;
+import org.openide.util.Lookup;
```
Comment (JaroslavTulach) [inline platform/openide.util.ui/src/org/netbeans/modules/openide/util/NbMutexEventProvider.java:27]: Write a test for `Mutex.EVENT` interaction.

**Review by JaroslavTulach — CHANGES_REQUESTED** [review]

**Comment by JaroslavTulach** [comment]
Let's leave this for 12.3 - it requires an API change from #2654 and even after that it may be a bit buggy: I have seen the "Run priming build" to remain after running the priming build (CCing @lkishalmi) - e.g. the dialog was displayed again and again. When I tested this in NetBeans, the behavior was the same - after running priming build the task to run priming build remained in the list.

File: platform/openide.util.ui/src/org/netbeans/modules/openide/util/NbMutexEventProvider.java (hunk 1 above)
Comment (sdedic) [reply platform/openide.util.ui/src/org/netbeans/modules/openide/util/NbMutexEventProvider.java:27]: @jtulach is this note still relevant after 1b68acdc17 ?

**Review by JaroslavTulach — APPROVED** [review]
Approving for 12.3

**Comment by sdedic** [comment]
rebased on latest master.

**Pull request merged**